File: 2017/02/corruption_checksum.go
package main

import (
	"fmt"
	"io/ioutil"
	"math"
	"strconv"
	"strings"
)

func readFile(filename string) (numbers [][]int) {
	dat, _ := ioutil.ReadFile("input")
	lines := strings.Split(string(dat), "\n")
	for _, line := range lines {
		numberStrings := strings.Split(line, "\t")
		numberInts := make([]int, len(numberStrings))
		for i, s := range numberStrings {
			numberInts[i], _ = strconv.Atoi(s)
		}
		numbers = append(numbers, numberInts)
	}
	return
}

// MinMax get both min and max of an int slice
func MinMax(ns []int) (min, max int) {
	min = math.MaxInt64
	for _, n := range ns {
		if n < min {
			min = n
		}
		if n > max {
			max = n
		}
	}
	return
}

func main() {
	numbers := readFile("input")

	var total, minN, maxN int
	for _, line := range numbers {
		minN, maxN = MinMax(line)
		total += maxN - minN
	}
	fmt.Println(total)

	total = 0
	for _, line := range numbers {
		for i := range line {
			for j := range line {
				if i != j && line[i]%line[j] == 0 {
					total += line[i] / line[j]
				}
			}
		}
	}
	fmt.Println(total)
}

File: 2017/02/corruption_checksum.py
##!/usr/bin/env python3

with open("input") as f:
    lines = f.readlines()

total = 0
for line in lines:
    items = [int(x) for x in line.split()]
    total += max(items) - min(items)

print(total)

total = 0
for line in lines:
    items = sorted([int(x) for x in line.split()], reverse=True)
    for i in range(len(items)):
        for j in range(i+1, len(items)):
            if items[i] % items[j] == 0:
                total += items[i] / items[j]
print(int(total))